src/main/psxsdk/libgpu/sys.c

```diff
@@ -319,9 +326,31 @@ u_long get_tw(RECT* arg0) {
 
 INCLUDE_ASM("main/nonmatchings/psxsdk/libgpu/sys", get_dx);
 
-s32 _status(void) { return *D_8002C280; }
+s32 _status(void) { return *GPU_STATUS; }
+
+s32 _otc(s32 arg0, s32 arg1) {
+    s32 temp_a0;
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libgpu/sys", _otc);
+    *DPCR |= 0x08000000;
+    *DMA6_CHCR = 0;
+    temp_a0 = arg0 - 4 + arg1 * 4;
+    *DMA6_MADR = temp_a0;
+    *DMA6_BCR = arg1;
+    *DMA6_CHCR = 0x11000002;
+    set_alarm();
+    if (*DMA6_CHCR & 0x01000000) {
+        while (1) {
+            if (get_alarm()) {
+                return -1;
+            } else {
+                if (!(*DMA6_CHCR & (1 << 24))) {
+                    break;
+                }
+            }
```

Comment:
```suggestion
            } else if (!(*DMA6_CHCR & (1 << 24))) {
                break;
            }
```
maybe like this is a bit more compact

Reply:
Done